### DISCUSSION TITLE
Coreset indices increasing when training images with patchcore from anomalib

### DISCUSSION THREAD

**Discussion by cory0619** [open-edge-platform/anomalib#1645] [Q&A] [answered]
Hello!
I'm trying to train images with PatchCore from anomalib.
but, Coreset incdies increasing when training images on each epoch like 22892, 45785, 68676, 91571 ...

Is it normal?

Thanks a lot in advance.

**Comment by alexriedel1** [comment]
can you please show some console output or describe exactly what the problem is? 
If you mean the index selection function, that outputs something like this
`Selecting Coreset Indices.:  95%|█████████▌| 33571/35251 [04:43<00:14, 118.24it/s]`
it's expected behaviour for the numbers to increase

**Reply by cory0619** [reply]
@alexriedel1 Thanks for the reply.
That's correct. I modified max_epochs: 1 as 500 so, as the number of epochs increased, 33571/35251 almost doubled each time. Finally, I set max_epochs as 1.

**Comment by abc-125** [comment, marked as answer]
Hello, PatchCore is supposed to be trained for one epoch only, as described in their [paper](https://arxiv.org/pdf/2106.08265.pdf). It extracts features from the training data to create a memory bank, which is used for comparison with the new input. To fix the number of epochs, you need to set `max_epochs: 1` in the config file.

**Reply by cory0619** [reply]
@abc-125 
Thanks for the reply.
I modified max_epochs: 1 as 500... I thought models like Mask-RCNN or UNet inference through iterations of epochs. Thank you.

**Reply by abc-125** [reply]
You're welcome! If you are interested, check their paper, it is quite unusual subclass of models, based on the memorization of features instead of typical training 😄
